Control/public/services/DetectorService.js

```diff
@@ -26,6 +26,9 @@ export default class DetectorService extends Observable {
   constructor(model) {
     super();
     this.model = model;
+    this.roles = this.model.session.access
```

Comment:
In the constructor usually we initialize with empty values and then any logic is applied in the `init` method.
Could you please define roles as empty array and move the logic to init method?